lib/ref.rb

```diff
@@ -18,12 +18,6 @@ module Ref
     if defined?(RUBY_ENGINE) && RUBY_ENGINE == 'rbx'
       # If using Rubinius set the implementation to use WeakRef since it is very efficient and using finalizers is not.
       require 'ref/weak_reference/weak_ref'
-    elsif defined?(::ObjectSpace::WeakMap)
-      # Ruby 2.0 has a working implementation of weakref.rb backed by the new ObjectSpace::WeakMap
-      require 'ref/weak_reference/weak_ref'
-    elsif defined?(::ObjectSpace._id2ref)
-      # If ObjectSpace can lookup objects from their object_id, then use the pure ruby implementation.
-      require 'ref/weak_reference/pure_ruby'
```

Comment:
`pure_ruby` was being used only when running it on MRI 1.8 and we're dropping 1.8 support.